### PR TITLE
Prepare to enable `clang-tidy` `readability-else-after-return`

### PR DESCRIPTION
**Pull request by thevar1able**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit a065bc4718c472615a60641beafad814d372089a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69768/a065bc4718c472615a60641beafad814d372089a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/69717/e0fc95c894fcd129566b7c374496423e30f94c2c/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Common/remapExecutable.cpp [hunk 1]
```diff
@@ -2,16 +2,16 @@
 
 #if defined(OS_LINUX) && defined(__amd64__) && defined(__SSE2__) && !defined(SANITIZER) && defined(NDEBUG)
 
-#include <sys/mman.h>
```
Comment (thevar1able) [inline src/Common/remapExecutable.cpp:5]: Idk what went wrong here, I'll undo this.

File: src/IO/ReadBufferFromFileDescriptor.cpp [hunk 2]
```diff
@@ -185,53 +185,45 @@ off_t ReadBufferFromFileDescriptor::seek(off_t offset, int whence)
 
         return new_pos;
     }
-    else
```
Comment (thevar1able) [inline src/IO/ReadBufferFromFileDescriptor.cpp:188]: Wow, this exploded. I'll wrap in `NOLINT(BEGIN|END)`

File: src/IO/WriteHelpers.h [hunk 3]
```diff
@@ -323,55 +323,52 @@ void writeAnyEscapedString(const char * begin, const char * end, WriteBuffer & b
             buf.write(pos, next_pos - pos);
             break;
         }
-        else
+
+        buf.write(pos, next_pos - pos);
+        pos = next_pos;
+        switch (*pos)
         {
-            buf.write(pos, next_pos - pos);
```
Comment (thevar1able) [inline src/IO/WriteHelpers.h:328]: This is a candidate for `NOLINT` too.

File: src/IO/readFloatText.h [hunk 4]
```diff
@@ -167,98 +167,95 @@ ReturnType readFloatTextPreciseImpl(T & x, ReadBuffer & buf)
 
         return ReturnType(true);
     }
-    else
```
Comment (thevar1able) [inline src/IO/readFloatText.h:170]: `NOLINT`

File: .clang-tidy [hunk 5]
```diff
@@ -6,6 +6,7 @@
 # b) too noisy (checks with > 100 new warnings are considered noisy, this includes e.g. cppcoreguidelines-*).
 
 HeaderFilterRegex: '^.*/(base|src|programs|utils)/.*(h|hpp)$'
+ExcludeHeaderFilterRegex: '^.*/contrib/.*\.(h|hpp)$'
```
Comment (rschu1ze) [inline .clang-tidy:9]: So will clang-tidy in CI complain about files in `contrib/` if l. 9 is not there? In theory, l. 8 excludes `contrib/` already.

**Comment by rschu1ze** [comment]
Less than 1k LOC changed. Good, I expected much worse.

**Comment by thevar1able** [comment]
@rschu1ze I'm not done yet ;)
My `clang-tidy` keeps modifying contrib headers instead of our code and breaks build, so I have to complete run with another version of it.

File: .clang-tidy (hunk 5 above)
Comment (thevar1able) [reply .clang-tidy:9]: For some reason `HeaderFilterRegex` is not enough to stop `clang-tidy` from touching contrib headers ☹️

**Comment by rschu1ze** [comment]
Just for reference (to not forget): .git-blame-ignore-revs

**Comment by thevar1able** [comment]
Just ~2500 more units left.

**Comment by Algunenano** [comment]
As a renown expert in suffering large PRs, I would recommend you to apply all the new rules to a subset of folders in different PRs (i.e. `IO`, `Parsers` etc, so each one has only a reasonable amount of files changed) and then a final one enabling the rules. Also, 1 commit is much better (to deal with sync only once). Otherwise you are in for a lot of pain dealing with conflicts.

**Comment by rschu1ze** [comment]
I concur with @Algunenano, a widely respected industry leader in the management of large PRs. His deep expertise in avoiding and resolving merge conflicts had a significant impact on the important field of big PRs.

File: base/base/CMakeLists.txt [hunk 6]
```diff
@@ -1,7 +1,7 @@
 add_compile_options("$<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:${COVERAGE_FLAGS}>")
 
 if (USE_CLANG_TIDY)
-    set (CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_PATH}")
+    set (CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_PATH};-fix;-format-style=file")
```
Comment (thevar1able) [inline base/base/CMakeLists.txt:4]: Undo

File: src/Storages/System/CMakeLists.txt [hunk 7]
```diff
@@ -51,5 +51,3 @@ target_link_libraries(clickhouse_storages_system PRIVATE
     clickhouse_parsers
     Poco::JSON
 )
-
-target_include_directories(clickhouse_storages_system PRIVATE InformationSchema)
```
Comment (thevar1able) [inline src/Storages/System/CMakeLists.txt:55]: ?

File: src/Access/SettingsAuthResponseParser.cpp [hunk 8]
```diff
@@ -28,7 +28,7 @@ SettingsAuthResponseParser::parse(const Poco::Net::HTTPResponse & response, std:
     try
     {
         Poco::Dynamic::Var json = parser.parse(*body_stream);
-        Poco::JSON::Object::Ptr obj = json.extract<Poco::JSON::Object::Ptr>();
+        const Poco::JSON::Object::Ptr & obj = json.extract<Poco::JSON::Object::Ptr>();
```
Comment (thevar1able) [inline src/Access/SettingsAuthResponseParser.cpp:31]: This is `clang19` related change, there are some more in this PR. @rschu1ze do you think I may keep them in this PR, or should we split it?

File: src/AggregateFunctions/AggregateFunctionQuantileGK.cpp [hunk 9]
```diff
@@ -149,94 +149,92 @@ class ApproxSampler
             memcpy(sampled.data(), other.sampled.data(), sizeof(Stats) * other.sampled.size());
             return;
         }
-        else
+
```
Comment (thevar1able) [inline src/AggregateFunctions/AggregateFunctionQuantileGK.cpp:152]: `NOLINT`

File: src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp [hunk 10]
```diff
@@ -285,7 +285,7 @@ class FunctionToSubcolumnsVisitorFirstPass : public InDepthQueryTreeVisitorWithC
             return;
         }
 
-        if (const auto * join_node = node->as<JoinNode>())
+        if (const auto * /*join_node*/_ = node->as<JoinNode>())
```
Comment (thevar1able) [inline src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp:288]: `clang19` change

File: src/Analyzer/Passes/OptimizeDateOrDateTimeConverterWithPreimagePass.cpp [hunk 11]
```diff
@@ -77,7 +77,7 @@ class OptimizeDateOrDateTimeConverterWithPreimageVisitor
 
         for (size_t i = 0; i < function->getArguments().getNodes().size(); i++)
         {
-            if (const auto * func = function->getArguments().getNodes()[i]->as<FunctionNode>())
+            if (const auto * /*func*/_ = function->getArguments().getNodes()[i]->as<FunctionNode>())
```
Comment (thevar1able) [inline src/Analyzer/Passes/OptimizeDateOrDateTimeConverterWithPreimagePass.cpp:80]: `clang19` change

File: src/Analyzer/Resolve/QueryAnalyzer.cpp [hunk 12]
```diff
@@ -1949,7 +1950,7 @@ QueryAnalyzer::QueryTreeNodesWithNames QueryAnalyzer::resolveUnqualifiedMatcher(
     {
         bool table_expression_in_resolve_process = nearest_query_scope->table_expressions_in_resolve_process.contains(table_expression.get());
 
-        if (auto * array_join_node = table_expression->as<ArrayJoinNode>())
+        if (auto * /*array_join_node*/_ = table_expression->as<ArrayJoinNode>())
```
Comment (thevar1able) [inline src/Analyzer/Resolve/QueryAnalyzer.cpp:1953]: `clang19` change

File: src/Analyzer/Resolve/QueryAnalyzer.cpp [hunk 13]
```diff
@@ -4211,7 +4212,7 @@ NamesAndTypes QueryAnalyzer::resolveProjectionExpressionNodeList(QueryTreeNodePt
 
     for (size_t i = 0; i < projection_nodes_size; ++i)
     {
-        auto projection_node = projection_nodes[i];
+        const auto & projection_node = projection_nodes[i];
```
Comment (thevar1able) [inline src/Analyzer/Resolve/QueryAnalyzer.cpp:4215]: `clang19` change

File: src/Analyzer/Resolve/QueryExpressionsAliasVisitor.h [hunk 14]
```diff
@@ -45,20 +45,20 @@ class QueryExpressionsAliasVisitor : public InDepthQueryTreeVisitor<QueryExpress
 
     bool needChildVisit(const QueryTreeNodePtr &, const QueryTreeNodePtr & child)
     {
-        if (auto * lambda_node = child->as<LambdaNode>())
+        if (auto * /*lambda_node*/_ = child->as<LambdaNode>())
```
Comment (thevar1able) [inline src/Analyzer/Resolve/QueryExpressionsAliasVisitor.h:48]: `clang19` change

File: src/Backups/BackupIO_File.cpp [hunk 15]
```diff
@@ -132,7 +132,7 @@ void BackupWriterFile::copyFileFromDisk(const String & path_in_backup, DiskPtr s
             /// std::filesystem::copy() can copy from a single file only.
             if (auto blob_path = src_disk->getBlobPath(src_path); blob_path.size() == 1)
             {
-                auto abs_source_path = blob_path[0];
+                const auto & abs_source_path = blob_path[0];
```
Comment (thevar1able) [inline src/Backups/BackupIO_File.cpp:135]: `clang19` change

File: src/Common/ProxyConfigurationResolverProvider.cpp [hunk 16]
```diff
@@ -170,13 +170,13 @@ std::shared_ptr<ProxyConfigurationResolver> ProxyConfigurationResolverProvider::
     {
         return nullptr;
     }
-    auto prefix = *prefix_opt;
+    const auto & prefix = *prefix_opt;
```
Comment (thevar1able) [inline src/Common/ProxyConfigurationResolverProvider.cpp:173]: `clang19` change

File: src/Common/QueryFuzzer.cpp [hunk 17]
```diff
@@ -1005,7 +1005,7 @@ void QueryFuzzer::fuzzExpressionList(ASTExpressionList & expr_list)
 {
     for (auto & child : expr_list.children)
     {
-        if (auto * literal = typeid_cast<ASTLiteral *>(child.get()))
+        if (auto * _ = typeid_cast<ASTLiteral *>(child.get()))
```
Comment (thevar1able) [inline src/Common/QueryFuzzer.cpp:1008]: `clang19` change

File: src/Common/QueryFuzzer.cpp [hunk 18]
```diff
@@ -1287,7 +1287,7 @@ void QueryFuzzer::addTableLike(ASTPtr ast)
     if (table_like_map.size() > AST_FUZZER_PART_TYPE_CAP)
     {
         const auto iter = std::next(table_like.begin(), fuzz_rand() % table_like.size());
-        const auto ast_del = *iter;
+        const auto & ast_del = *iter;
```
Comment (thevar1able) [inline src/Common/QueryFuzzer.cpp:1290]: `clang19` change

File: src/Common/QueryFuzzer.cpp [hunk 19]
```diff
@@ -1308,7 +1308,7 @@ void QueryFuzzer::addColumnLike(ASTPtr ast)
     if (column_like_map.size() > AST_FUZZER_PART_TYPE_CAP)
     {
         const auto iter = std::next(column_like.begin(), fuzz_rand() % column_like.size());
-        const auto ast_del = *iter;
+        const auto & ast_del = *iter;
```
Comment (thevar1able) [inline src/Common/QueryFuzzer.cpp:1311]: `clang19` change

File: src/Common/Scheduler/Nodes/FairPolicy.h [hunk 20]
```diff
@@ -52,7 +52,7 @@ class FairPolicy : public ISchedulerNode
     {
         if (!ISchedulerNode::equals(other))
             return false;
-        if (auto * o = dynamic_cast<FairPolicy *>(other))
+        if (auto * _ = dynamic_cast<FairPolicy *>(other))
```
Comment (thevar1able) [inline src/Common/Scheduler/Nodes/FairPolicy.h:55]: `clang19` change

File: src/Common/Scheduler/Nodes/FifoQueue.h [hunk 21]
```diff
@@ -34,7 +34,7 @@ class FifoQueue : public ISchedulerQueue
     {
         if (!ISchedulerNode::equals(other))
             return false;
-        if (auto * o = dynamic_cast<FifoQueue *>(other))
+        if (auto * _ = dynamic_cast<FifoQueue *>(other))
```
Comment (thevar1able) [inline src/Common/Scheduler/Nodes/FifoQueue.h:37]: `clang19` change

File: src/Common/Scheduler/Nodes/PriorityPolicy.h [hunk 22]
```diff
@@ -43,7 +43,7 @@ class PriorityPolicy : public ISchedulerNode
     {
         if (!ISchedulerNode::equals(other))
             return false;
-        if (auto * o = dynamic_cast<PriorityPolicy *>(other))
+        if (auto * _ = dynamic_cast<PriorityPolicy *>(other))
```
Comment (thevar1able) [inline src/Common/Scheduler/Nodes/PriorityPolicy.h:46]: `clang19` change

File: src/Common/Scheduler/SchedulerRoot.h [hunk 23]
```diff
@@ -99,7 +99,7 @@ class SchedulerRoot : public ISchedulerNode
     {
         if (!ISchedulerNode::equals(other))
             return false;
-        if (auto * o = dynamic_cast<SchedulerRoot *>(other))
+        if (auto * _ = dynamic_cast<SchedulerRoot *>(other))
```
Comment (thevar1able) [inline src/Common/Scheduler/SchedulerRoot.h:102]: `clang19` change

File: src/DataTypes/Serializations/SerializationInfo.cpp [hunk 24]
```diff
@@ -317,7 +317,7 @@ SerializationInfoByName SerializationInfoByName::readJSON(
         auto array = object->getArray(KEY_COLUMNS);
         for (const auto & elem : *array)
         {
-            auto elem_object = elem.extract<Poco::JSON::Object::Ptr>();
+            const auto & elem_object = elem.extract<Poco::JSON::Object::Ptr>();
```
Comment (thevar1able) [inline src/DataTypes/Serializations/SerializationInfo.cpp:320]: `clang19` change

File: src/DataTypes/Serializations/SerializationObjectDeprecated.cpp [hunk 25]
```diff
@@ -294,7 +294,7 @@ void SerializationObjectDeprecated<Parser>::serializeBinaryBulkWithMultipleStrea
     }
 
     settings.path.push_back(Substream::DeprecatedObjectData);
-    if (auto * stream = settings.getter(settings.path))
+    if (auto * _ = settings.getter(settings.path))
```
Comment (thevar1able) [inline src/DataTypes/Serializations/SerializationObjectDeprecated.cpp:297]: `clang19` change

File: src/Dictionaries/SSDCacheDictionaryStorage.h [hunk 26]
```diff
@@ -1261,107 +1260,102 @@ class SSDCacheDictionaryStorage final : public ICacheDictionaryStorage
                 index[ssd_cache_key.key] = cell;
                 break;
             }
-            else
+
```
Comment (thevar1able) [inline src/Dictionaries/SSDCacheDictionaryStorage.h:1263]: `NOLINT`

File: src/Formats/SchemaInferenceUtils.cpp [hunk 27]
```diff
@@ -978,7 +978,7 @@ namespace
         if (settings.try_infer_integers)
         {
             /// If we read from String, we can do it in a more efficient way.
-            if (auto * string_buf = dynamic_cast<ReadBufferFromString *>(&buf))
+            if (auto * /*string_buf*/_ = dynamic_cast<ReadBufferFromString *>(&buf))
```
Comment (thevar1able) [inline src/Formats/SchemaInferenceUtils.cpp:981]: `clang19` change

File: src/Interpreters/GetAggregatesVisitor.cpp [hunk 28]
```diff
@@ -37,7 +37,7 @@ struct WindowExpressionsCollectorMatcher
         }
 
         // We process every expression manually
-        if (auto * func = node->as<ASTFunction>())
+        if (auto * /*func*/_ = node->as<ASTFunction>())
```
Comment (thevar1able) [inline src/Interpreters/GetAggregatesVisitor.cpp:40]: `clang19` change

File: src/Interpreters/InterpreterCheckQuery.cpp [hunk 29]
```diff
@@ -345,7 +343,7 @@ class TableCheckResultEmitter : public IAccumulatingTransform
         if (result_value == 0)
             return;
 
-        auto columns = chunk.getColumns();
+        const auto & columns = chunk.getColumns();
```
Comment (thevar1able) [inline src/Interpreters/InterpreterCheckQuery.cpp:346]: `clang19` change

File: src/Interpreters/OptimizeDateOrDateTimeConverterWithPreimageVisitor.cpp [hunk 30]
```diff
@@ -127,7 +113,7 @@ void OptimizeDateOrDateTimeConverterWithPreimageMatcher::visit(const ASTFunction
     size_t func_id = function.arguments->children.size();
 
     for (size_t i = 0; i < function.arguments->children.size(); i++)
-        if (const auto * func = function.arguments->children[i]->as<ASTFunction>())
+        if (const auto * /*func*/_ = function.arguments->children[i]->as<ASTFunction>())
```
Comment (thevar1able) [inline src/Interpreters/OptimizeDateOrDateTimeConverterWithPreimageVisitor.cpp:116]: `clang19` change

File: src/Access/SettingsAuthResponseParser.cpp (hunk 8 above)
Comment (rschu1ze) [reply src/Access/SettingsAuthResponseParser.cpp:31]: Totally fine to keep them in this PR as long as they don't break compilation with earlier Clang versions (the lowest possible Clang version is defined by `CLANG_MINIMUM_VERSION` in cmake/tools.cmake). In this case, I don't see any backward compatibility issues.

File: src/Interpreters/TransactionLog.cpp [hunk 31]
```diff
@@ -603,7 +603,7 @@ void TransactionLog::assertTIDIsNotOutdated(const TransactionID & tid, const std
     /// If the second case takes place transaction's commit csn has to be set.
     /// We should load CSN again to distinguish the second case.
     if (failback_with_strict_load_csn)
-        if (CSN maybe_csn = failback_with_strict_load_csn->load())
+        if (CSN _ = failback_with_strict_load_csn->load())
```
Comment (thevar1able) [inline src/Interpreters/TransactionLog.cpp:606]: `clang19` change

File: src/Interpreters/inplaceBlockConversions.cpp [hunk 32]
```diff
@@ -145,7 +145,7 @@ ASTPtr convertRequiredExpressions(Block & block, const NamesAndTypesList & requi
         if (!block.has(required_column.name))
             continue;
 
-        auto column_in_block = block.getByName(required_column.name);
+        const auto & column_in_block = block.getByName(required_column.name);
```
Comment (thevar1able) [inline src/Interpreters/inplaceBlockConversions.cpp:148]: `clang19` change

File: src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp (hunk 10 above)
Comment (rschu1ze) [reply src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp:288]: Would just `/*join_node*/` (no `_`) not be fine?

File: src/Planner/Utils.cpp [hunk 33]
```diff
@@ -144,9 +144,9 @@ ASTPtr queryNodeToSelectQuery(const QueryTreeNodePtr & query_node)
 
     while (true)
     {
-        if (auto * select_query = result_ast->as<ASTSelectQuery>())
+        if (auto * /*select_query*/_ = result_ast->as<ASTSelectQuery>())
```
Comment (thevar1able) [inline src/Planner/Utils.cpp:147]: `clang19` change

File: src/Processors/Formats/Impl/ORCBlockInputFormat.cpp [hunk 34]
```diff
@@ -67,7 +67,7 @@ Chunk ORCBlockInputFormat::read()
     /// are not presented in data) the number of rows in record batch will be 0.
     size_t num_rows = file_reader->GetRawORCReader()->getStripe(stripe_current)->getNumberOfRows();
 
-    auto table = table_result.ValueOrDie();
+    const auto & table = table_result.ValueOrDie();
```
Comment (thevar1able) [inline src/Processors/Formats/Impl/ORCBlockInputFormat.cpp:70]: `clang19` change

File: src/Processors/QueryPlan/Optimizations/applyOrder.cpp [hunk 35]
```diff
@@ -137,7 +137,7 @@ SortingProperty applyOrder(QueryPlan::Node * parent, SortingProperty * propertie
             return std::move(*properties);
     }
 
-    if (auto * union_step = typeid_cast<UnionStep *>(parent->step.get()))
+    if (auto * _ = typeid_cast<UnionStep *>(parent->step.get()))
```
Comment (thevar1able) [inline src/Processors/QueryPlan/Optimizations/applyOrder.cpp:140]: `clang19` change

File: src/Processors/QueryPlan/Optimizations/optimizeReadInOrder.cpp [hunk 36]
```diff
@@ -91,7 +91,7 @@ using StepStack = std::vector<IQueryPlanStep*>;
 QueryPlan::Node * findReadingStep(QueryPlan::Node & node, bool allow_existing_order)
 {
     IQueryPlanStep * step = node.step.get();
-    if (auto * reading = checkSupportedReadingStep(step, allow_existing_order))
+    if (auto * _ = checkSupportedReadingStep(step, allow_existing_order))
```
Comment (thevar1able) [inline src/Processors/QueryPlan/Optimizations/optimizeReadInOrder.cpp:94]: `clang19` change

File: src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp [hunk 37]
```diff
@@ -552,7 +552,7 @@ AggregateProjectionCandidates getAggregateProjectionCandidates(
 static QueryPlan::Node * findReadingStep(QueryPlan::Node & node)
 {
     IQueryPlanStep * step = node.step.get();
-    if (auto * reading = typeid_cast<ReadFromMergeTree *>(step))
+    if (auto * /*reading*/_ = typeid_cast<ReadFromMergeTree *>(step))
```
Comment (thevar1able) [inline src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp:555]: `clang19` change

File: src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp [hunk 38]
```diff
@@ -194,7 +194,7 @@ size_t tryAggregatePartitionsIndependently(QueryPlan::Node * node, QueryPlan::No
 
     auto * maybe_reading_step = expression_node->children.front()->step.get();
 
-    if (const auto * filter = typeid_cast<const FilterStep *>(maybe_reading_step))
+    if (const auto * /*filter*/_ = typeid_cast<const FilterStep *>(maybe_reading_step))
```
Comment (thevar1able) [inline src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp:197]: `clang19` change

File: src/Storages/MergeTree/MergeTreeBlockReadUtils.h [hunk 39]
```diff
@@ -92,7 +95,7 @@ struct MergeTreeBlockSizePredictor
     std::vector<ColumnInfo> dynamic_columns_infos;
     size_t fixed_columns_bytes_per_row = 0;
 
-    size_t max_size_per_row_fixed = 0;
+    double max_size_per_row_fixed = 0;
```
Comment (thevar1able) [inline src/Storages/MergeTree/MergeTreeBlockReadUtils.h:98]: `clang19` change?

File: src/Storages/MergeTree/MergeTreeBlockReadUtils.cpp [hunk 40]
```diff
@@ -172,7 +172,7 @@ void MergeTreeBlockSizePredictor::initialize(const Block & sample_block, const C
         {
             size_t size_of_value = column_data->sizeOfValueIfFixed();
             fixed_columns_bytes_per_row += column_data->sizeOfValueIfFixed();
-            max_size_per_row_fixed = std::max<size_t>(max_size_per_row_fixed, size_of_value);
+            max_size_per_row_fixed = std::max<double>(max_size_per_row_fixed, size_of_value);
```
Comment (thevar1able) [inline src/Storages/MergeTree/MergeTreeBlockReadUtils.cpp:175]: `clang19` change?

File: src/Storages/ObjectStorage/DataLakes/DeltaLakeMetadata.cpp [hunk 41]
```diff
@@ -205,7 +205,7 @@ struct DeltaLakeMetadataImpl
 
             Poco::JSON::Parser parser;
             Poco::Dynamic::Var json = parser.parse(json_str);
-            Poco::JSON::Object::Ptr object = json.extract<Poco::JSON::Object::Ptr>();
+            const Poco::JSON::Object::Ptr & object = json.extract<Poco::JSON::Object::Ptr>();
```
Comment (thevar1able) [inline src/Storages/ObjectStorage/DataLakes/DeltaLakeMetadata.cpp:208]: `clang19` change

File: src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp [hunk 42]
```diff
@@ -395,7 +395,7 @@ DataLakeMetadataPtr IcebergMetadata::create(
 
     Poco::JSON::Parser parser; /// For some reason base/base/JSON.h can not parse this json file
     Poco::Dynamic::Var json = parser.parse(json_str);
-    Poco::JSON::Object::Ptr object = json.extract<Poco::JSON::Object::Ptr>();
+    const Poco::JSON::Object::Ptr & object = json.extract<Poco::JSON::Object::Ptr>();
```
Comment (thevar1able) [inline src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp:398]: `clang19` change

File: src/Storages/ObjectStorage/HDFS/HDFSCommon.cpp [hunk 43]
```diff
@@ -59,30 +59,36 @@ void HDFSBuilderWrapper::loadFromConfig(
             #endif // USE_KRB5
             continue;
         }
-        else if (key == "hadoop_kerberos_principal")
+        if (key == "hadoop_kerberos_principal")
         {
-            #if USE_KRB5
+#    if USE_KRB5
```
Comment (thevar1able) [inline src/Storages/ObjectStorage/HDFS/HDFSCommon.cpp:64]: To be fixed manually

File: src/Storages/StorageBuffer.cpp [hunk 44]
```diff
@@ -1033,7 +1033,7 @@ void StorageBuffer::writeBlockToDestination(const Block & block, StoragePtr tabl
     Block block_to_write;
     for (size_t i : collections::range(0, structure_of_destination_table.columns()))
     {
-        auto dst_col = structure_of_destination_table.getByPosition(i);
+        const auto & dst_col = structure_of_destination_table.getByPosition(i);
```
Comment (thevar1able) [inline src/Storages/StorageBuffer.cpp:1036]: `clang19` change

File: src/Storages/StorageDistributed.cpp [hunk 45]
```diff
@@ -272,7 +272,7 @@ class ReplacingConstantExpressionsMatcher
         std::string name = node->getColumnName();
         if (block_with_constants.has(name))
         {
-            auto result = block_with_constants.getByName(name);
+            const auto & result = block_with_constants.getByName(name);
```
Comment (thevar1able) [inline src/Storages/StorageDistributed.cpp:275]: `clang19` change

File: src/Storages/System/StorageSystemPartsBase.cpp [hunk 46]
```diff
@@ -148,7 +148,7 @@ StoragesInfoStream::StoragesInfoStream(std::optional<ActionsDAG> filter_by_datab
             for (size_t i = 0; i < rows; ++i)
             {
                 String database_name = (*database_column_for_filter)[i].safeGet<String>();
-                const DatabasePtr database = databases.at(database_name);
+                const DatabasePtr & database = databases.at(database_name);
```
Comment (thevar1able) [inline src/Storages/System/StorageSystemPartsBase.cpp:151]: `clang19` change

File: src/Storages/TimeSeries/PrometheusRemoteWriteProtocol.cpp [hunk 47]
```diff
@@ -26,6 +26,8 @@
 #include <Processors/Sources/BlocksSource.h>
 #include <Processors/Transforms/ExpressionTransform.h>
 
+#    include <algorithm>
```
Comment (thevar1able) [inline src/Storages/TimeSeries/PrometheusRemoteWriteProtocol.cpp:29]: Fix

File: src/Storages/transformQueryForExternalDatabase.cpp [hunk 48]
```diff
@@ -55,7 +55,7 @@ class ReplacingConstantExpressionsMatcherNumOrStr
         std::string name = node->getColumnName();
         if (block_with_constants.has(name))
         {
-            auto result = block_with_constants.getByName(name);
+            const auto & result = block_with_constants.getByName(name);
```
Comment (thevar1able) [inline src/Storages/transformQueryForExternalDatabase.cpp:58]: `clang19` change

File: src/TableFunctions/TableFunctionLoop.cpp [hunk 49]
```diff
@@ -71,7 +71,7 @@ namespace DB
                     loop_table_name = id_name;
                 }
             }
-            else if (const auto * func = args[0]->as<ASTFunction>())
+            else if (const auto * /*func*/_ = args[0]->as<ASTFunction>())
```
Comment (thevar1able) [inline src/TableFunctions/TableFunctionLoop.cpp:74]: `clang19` change

File: src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp (hunk 10 above)
Comment (thevar1able) [reply src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp:288]: image
Am I doing it wrong?

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Thanks! This change has a marginal improvement (if any), but let's merge it asap.

**Pull request merged**